pkg/controller/syncconfig/handler.go

```diff
@@ -2,12 +2,105 @@ package syncconfig
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/vshn/espejo/pkg/apis/sync/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 func (r *ReconcileSyncConfig) handle(ctx context.Context, syncConfig *v1alpha1.SyncConfig, request reconcile.Request) error {
+	namespaces, err := r.getNamespaces(ctx, syncConfig, request)
```

Comment:
The cognitive complexity of this method is too hight. I suggest to break this down into several smaller functions.